src/addgrid.f

```diff
@@ -1,67 +1,47 @@
+!>    @file
+!>    @brief This subroutine packs up a Grid Definition Section
+!>    (Section 3) and adds it to a GRIB2 message.
+!>    @author Gilbert ORG: W/NP11 @date 2000-05-01
+!>
+!>    This routine is used with routines gribcreate(), addlocal(), addfield(),
+!>    and gribend() to create a complete GRIB2 message. Subroutine
+!>    gribcreate must be called first to initialize a new GRIB2 message.
+!>
+!>    @param[in] cgrib Character array to contain the GRIB2 message
+!>    @param[in] lcgrib Maximum length (bytes) of array cgrib.
+!>    @param[in] igds Contains information needed for GRIB Grid Definition Section 3.
+!>    Must be dimensioned >= 5.
+!>    - igds(1)=Source of grid definition (see Code Table 3.0)
+!>    - igds(2)=Number of grid points in the defined grid.
+!>    - igds(3)=Number of octets needed for each
+!>    additional grid points definition.Used to define number of
+!>    points in each row ( or column ) for non-regular grids. = 0, if using regular grid.
+!>    - igds(4)=Interpretation of list for optional points
+!>    definition.  (Code Table 3.11)
+!>    - igds(5)=Grid Definition Template Number (Code Table 3.1)
+!>    @param[in] igdstmpl Contains the data values for the specified Grid Definition
+!>    Template ( NN=igds(5) ).  Each element of this integer array
+!>    contains an entry (in the order specified) of Grid Defintion Template 3.NN
+!>    @param[in] igdstmplen Max dimension of igdstmpl()
+!>    @param[in] ideflist (Used if igds(3) .ne. 0)  This array contains the
+!>    number of grid points contained in each row ( or column )
+!>    @param[in] idefnum (Used if igds(3) .ne. 0)  The number of entries
+!>    in array ideflist.  i.e. number of rows ( or columns )
+!>    for which optional grid points are defined.
+!>    @param[out] ierr Error return code.
+!>    - 0 = no error
+!>    - 1 = GRIB message was not initialized.  Need to call routine gribcreate first.
+!>    - 2 = GRIB message already complete.  Cannot add new section.
+!>    - 3 = Sum of Section byte counts doesn't add to total byte count.
+!>    - 4 = Previous Section was not 1, 2 or 7.
+!>    - 5 = Could not find requested Grid Definition Template.
+!>
+!>    REMARKS: Note that the Local Use Section ( Section 2 ) can only follow
+!>          Section 1 or Section 7 in a GRIB2 message.
+!>
```

Comment:
Subroutine @author and @date tag needed here. Just a repeat of the one used in the file doxygen block.

Reply:
add it, Thanks!